### PR TITLE
[Simple] Add isArithmeticExpr() to expression

### PR DESCRIPTION
**Pull request by Aiee**
As title.

### CODE REVIEW

File: src/common/expression/Expression.h [hunk 1]
```diff
@@ -146,6 +146,10 @@ class Expression {
         return false;
     }
 
+    virtual bool isArithmeticExpr() const {
```
Comment (Shylock-Hg) [inline src/common/expression/Expression.h:149]: You could check it by kind_

Comment (Aiee) [reply src/common/expression/Expression.h:149]: There are multiple kinds of expressions that belong to arithmetic expressions, and every time when we need to check the expression type, we have to iterate all of them.

Comment (Shylock-Hg) [reply src/common/expression/Expression.h:149]: It's more friendly to cpu cache.

Comment (Aiee) [reply src/common/expression/Expression.h:149]: This function will only be called before the actual calculation of the expression and won't be involved in a hot spot.

**Review by yixinglu — APPROVED** [review]

**Review by nevermore3 — APPROVED** [review]

**Pull request merged**